types/mathjs/index.d.ts

```diff
@@ -1,2252 +1,4546 @@
+         */
+        eye(m: number, n: number, format?: string): Matrix | MathArray | number;
+
+        /**
+         * Filter the items in an array or one dimensional matrix.
+         * @param x A one dimensional matrix or array to filter
+         * @param test A function or regular expression to test items. All
+         * entries for which test returns true are returned. When test is a
+         * function, it is invoked with three parameters: the value of the
+         * element, the index of the element, and the matrix/array being
+         * traversed. The function must return a boolean.
+         */
+        filter(
+            x: Matrix | MathArray,
+            test: ((value:any, index: any, matrix: Matrix | MathArray)=>Matrix | MathArray) | RegExp
+        ): Matrix | MathArray;
+
+        /**
+         * Flatten a multi dimensional matrix into a single dimensional matrix.
+         * @param x Matrix to be flattened
+         * @returns Returns the flattened matrix
+         */
+        flatten(x: MathArray | Matrix): MathArray | Matrix;
+
+        /**
+         * Iterate over all elements of a matrix/array, and executes the given
+         * callback function.
+         * @param x The matrix to iterate on.
+         * @param callback The callback function is invoked with three
+         * parameters: the value of the element, the index of the element, and
+         * the Matrix/array being traversed.
+         */
+        forEach(x: Matrix | MathArray, callback: ((value:any, index: any, matrix: Matrix | MathArray)=>void)): void;
+
+        /**
+         * Calculate the inverse of a square matrix.
+         * @param x Matrix to be inversed
+         * @returns The inverse of x
+         */
+        inv(
+            x: number | Complex | MathArray | Matrix
+        ): number | Complex | MathArray | Matrix;
+
+        /**
+         * Calculate the kronecker product of two matrices or vectors
+         * @param x First vector
+         * @param y Second vector
+         * @returns Returns the kronecker product of x and y
+         */
+        kron(x: Matrix | MathArray, y: Matrix | MathArray): Matrix;
+
+        /**
+         * Iterate over all elements of a matrix/array, and executes the given
+         * callback function.
+         * @param x The matrix to iterate on.
+         * @param callback The callback function is invoked with three
+         * parameters: the value of the element, the index of the element, and
+         * the Matrix/array being traversed.
+         * @returns Transformed map of x
+         */
+        map(x: Matrix | MathArray, callback: ((value:any, index: any, matrix: Matrix | MathArray)=>Matrix | MathArray)): Matrix | MathArray;
+
+        /**
+         * Create a matrix filled with ones. The created matrix can have one or
+         * multiple dimensions.
+         * @param size The size of each dimension of the matrix
+         * @param format The matrix storage 
```

Comment:
I think these parameters should be optional: `rationalize(optional?: ..., detailed?: ...): ...`